core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -448,6 +448,15 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
         }
       }
     }
+
+    // System property spark.yarn.app.id must be set if user code ran by AM on a YARN cluster
+    // yarn-standalone is deprecated, but still supported
+    if ((get("spark.master") == "yarn-cluster" || get("spark.master") == "yarn-standalone") &&
```

Comment:
Oh my bad. I thought this was still in SparkContext. In fact, I think the check should be there. When I mentioned "check for `spark.yarn.app.id` in SparkConf" I didn't mean to change SparkConf, I meant to do as I suggested above (`!_conf.contains("spark.yarn.app.id"")`).